Framework/jmaqs-selenium/src/test/java/com/magenic/jmaqs/selenium/WebDriverFactoryTest.java

```diff
@@ -0,0 +1,274 @@
+      }
+    }
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getDefaultChromeOptionsTest() {
+    ChromeOptions options = WebDriverFactory.getDefaultChromeOptions();
+    Assert.assertNotNull(options);
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getDefaultHeadlessChromeOptionsTest() {
+    ChromeOptions options = WebDriverFactory.getDefaultHeadlessChromeOptions();
+    Assert.assertNotNull(options);
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getDefaultInternetExplorerOptionsTest() {
+    InternetExplorerOptions options = WebDriverFactory.getDefaultInternetExplorerOptions();
+    Assert.assertNotNull(options);
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getDefaultFirefoxOptionsTest() {
+    FirefoxOptions options = WebDriverFactory.getDefaultFirefoxOptions();
+    Assert.assertNotNull(options);
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getDefaultEdgeOptionsTest() {
+    EdgeOptions options = WebDriverFactory.getDefaultEdgeOptions();
+    Assert.assertNotNull(options);
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getChromeDriverTest() throws Exception {
+    ChromeDriver driver = null;
+    try {
+      driver = (ChromeDriver) WebDriverFactory.getBrowserWithDefaultConfiguration(BrowserType.Chrome);
+      Assert.assertNotNull(driver);
+    } finally {
+      if (driver != null) {
+        driver.quit();
+      }
+    }
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getHeadlessChromeDriverTest() throws Exception {
+    ChromeDriver driver = null;
+    try {
+      driver = (ChromeDriver) WebDriverFactory.getBrowserWithDefaultConfiguration(BrowserType.HeadlessChrome);
+      Assert.assertNotNull(driver);
+    } finally {
+      if (driver != null) {
+        driver.quit();
+      }
+    }
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getFirefoxDriverTest() throws Exception {
+    FirefoxDriver driver = null;
+    try {
+      driver = (FirefoxDriver) WebDriverFactory.getBrowserWithDefaultConfiguration(BrowserType.Firefox);
+      Assert.assertNotNull(driver);
+    } finally {
+      if (driver != null) {
+        driver.quit();
+      }
+    }
+  }
+
+  @Test(groups = TestCategories.Selenium)
+  public void getInternetExplorerDriverTest() throws Exception {
+    InternetExplorerDriver driver = null;
+    try {
+      driver = (InternetExplorerDriver) WebDriverFactory.getBrowserWithDefaultConfiguration(BrowserType.IE);
+      Assert.assertNotNull(driver);
+    } finally {
+      if (driver != null) {
+        driver.quit();
+      }
+    }
+  }
+
+  @Test(expectedExceptions = Exception.class, groups = TestCategories.Selenium)
+  public void getEdgeDriverTest() throws Exception {
+    EdgeDriver driver = null;
+    try {
+      driver = (EdgeDriver) WebDriverFactory.getBrowserWithDefaultConfiguration(BrowserType.Edge);
+      Assert.assertNotNull(driver);
+    } catch (Exception e) {
```

Comment:
Why are we catching the Exception when we are going directly to a finally? In this case you are just throwing it? We ight need this but the test might not be needed.

Reply:
I've removed this test. This isn't testable in our pipeline.